File: system/monolith/java/src/main/java/com/optivem/starter/monolith/core/dtos/PlaceOrderRequest.java
package com.optivem.starter.monolith.core.dtos;

import com.optivem.starter.monolith.core.validation.TypeValidationMessage;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Positive;
import lombok.Data;

@Data
public class PlaceOrderRequest {
    @NotBlank(message = "SKU must not be empty")
    private String sku;

    @NotNull(message = "Quantity must not be empty")
    @Positive(message = "Quantity must be positive")
    @TypeValidationMessage("Quantity must be an integer")
    private Integer quantity;
}

File: system/multitier/backend-dotnet/Core/Dtos/PlaceOrderRequest.cs
using System.ComponentModel.DataAnnotations;

namespace Optivem.Shop.Backend.Core.Dtos;

public class PlaceOrderRequest
{
    [Required(ErrorMessage = "SKU must not be empty")]
    public string? Sku { get; set; }

    [Required(ErrorMessage = "Quantity must not be empty")]
    [Range(1, int.MaxValue, ErrorMessage = "Quantity must be positive")]
    public int? Quantity { get; set; }
}
